projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -2377,6 +2377,22 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         }
     }
 
+    /**
+     * Returns the `IgxGridHeaderGroupComponent`'s minimum allloed width.
+     * Used internally for restricting header group component width.
+     * The values below depend on the header cell default right/left padding values.
+	 * @memberof IgxGridBaseComponent
+     */
+    get defaultHeaderGroupMinWidth(): number {
+        if (this.isCosy()) {
+            return 32;
+        } else if (this.isCompact()) {
+            return 24;
+        } else {
+            return 48;
+        }
+    }
+
```

Comment:
Do the listed sizes allow enough space so at least 1 character and the sorting icon is visible at all times?

Reply:
No, they only take header cell left and right padding into consideration. Sorting is considered only when resizing a column. I would rather leave it this way. I only want to guarantee that there will not be any cells misalignment on grid initialization, if the user has set set a column width that is smaller that the header paddings mentioned above.